build/Shared/SimplePool.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace NuGet
+{
+    internal class SimplePool<T> where T : class
```

Comment:
There's an impl of this class in array pool that can *probably* be removed now.

Reply:
That should have been removed in the new changes I pushed.

Reply:
I did a bad job in the conflict resolution. 

I tried to be helpful, lol :D 

You can make the change on your end, or I can follow up with another PR once the CI passes on this one.

Reply:
Done.